components/browser/session/src/main/java/mozilla/components/browser/session/manifest/WebAppManifest.kt

```diff
@@ -0,0 +1,165 @@
+ * Web app manifests are part of a collection of web technologies called progressive web apps, which are websites
+ * that can be installed to a device’s homescreen without an app store, along with other capabilities like working
+ * offline and receiving push notifications.
+ *
+ * https://developer.mozilla.org/en-US/docs/Web/Manifest
+ * https://www.w3.org/TR/appmanifest/
+ * https://developers.google.com/web/fundamentals/web-app-manifest/
+ *
+ * @property name Provides a human-readable name for the site when displayed to the user. For example, among a list of
+ * other applications or as a label for an icon.
+ * @property shortName Provides a short human-readable name for the application. This is intended for when there is
+ * insufficient space to display the full name of the web application, like device homescreens.
+ * @property startUrl The URL that loads when a user launches the application (e.g. when added to home screen),
+ * typically the index. Note that this has to be a relative URL, relative to the manifest url.
+ * @property display Defines the developers’ preferred display mode for the website.
+ * @property backgroundColor Defines the expected “background color” for the website. This value repeats what is
+ * already available in the site’s CSS, but can be used by browsers to draw the background color of a shortcut when
+ * the manifest is available before the stylesheet has loaded. This creates a smooth transition between launching the
+ * web application and loading the site's content.
+ * @property description Provides a general description of what the pinned website does.
+ * @property icons Specifies a list of image files that can serve as application icons, depending on context. For
+ * example, they can be used to represent the web application amongst a list of other applications, or to integrate the
+ * web application with an OS's task switcher and/or system preferences.
+ * @property dir Specifies the primary text direction for the name, short_name, and description members. Together with
+ * the lang member, it helps the correct display of right-to-left languages.
+ * @property lang Specifies the primary language for the values in the name and short_name members. This value is a
+ * string containing a single language tag (e.g. en-US).
+ * @property orientation Defines the default orientation for all the website's top level browsing contexts.
+ * @property scope Defines the navigation scope of this website's context. This restricts what web pages can be viewed
+ * while the manifest is applied. If the user navigates outside the scope, it returns to a normal web page inside a
+ * browser tab/window.
+ * @property themeColor Defines the default theme color for an application. This sometimes affects how the OS displays
+ * the site (e.g., on Android's task switcher, the theme color surrounds the site).
+ */
+data class WebAppManifest(
+    val name: String,
+    val startUrl: String,
+    val shortName: String? = null,
```

Comment:
That's how I had it initially. I changed the order to have the `required` values first. Otherwise you always have to specify the name for them. But then again no one is ever going to build that object manually, I guess. So I'm happy to change that too.

Reply:
> That's how I had it initially.

That's why in the comment has that order, I guess the order doesn't affect the final docs?

```kotlin
/**
 *@property name Provides a human-readable name for the site when displayed to the user. For example, among a list of
 * other applications or as a label for an icon.
 * @property shortName Provides a short human-readable name for the application. This is intended for when there is
 * insufficient space to display the full name of the web application, like device homescreens.
 * @property startUrl The URL that loads when a user launches the application (e.g. when added to home screen)
**/
```

Reply:
Yep! :)